providers/route53/route53Provider.go

```diff
@@ -305,3 +385,31 @@ func (r *route53Provider) EnsureDomainExists(domain string) error {
 	return err
 
 }
+
+func (r *route53Provider) waitUntilNameserversUpdate(operationId *string) error {
```

Comment:
I just tried this out with the CLI. I have no reason the believe the experience would be any different for the SDK.

I issued a `update-domain-nameserver` and then started watching the `get-domain-detail`. It still showed the old NS records until the operation actually completed successfully.

I also managed to get a failed operation when trying to set the NS records to unresolvable names. So it turns out it definitely is possible for operations to fail. It just gives you `FAILED` on the CLI - you need to check your domain contact's email address to figure out _why_ it failed. Gross. But more of a reason to ensure it succeeded before calling dnscontrol's job done.